pkg/controllers/v1alpha2/mpi_job_controller.go

```diff
@@ -486,6 +490,13 @@ func (c *MPIJobController) syncHandler(key string) error {
 		return err
 	}
 
+	if c.enableGangScheduling && done && isCleanUpPods(mpiJob.Spec.CleanPodPolicy) {
```

Comment:
If user set cleanupPodPolicy, will delete the podgroup alone with pods when train job have done.
@terrytangyuan  ptal :)